### PR TITLE
Add Support for Lilly58 r2g

### PR DESCRIPTION
**Pull request by e11i0t23**
## Description

Add support for the mechboards lily58 r2g, this is a revision of the lily58 which comes presolder featuring custom rgb and a base keymap with altered oled behaviour. SImilar also to the CRKBD R2G

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lily58/r2g/config.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+//#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+//#define LOCKING_RESYNC_ENABLE
+
+
```
Comment (fauxpark) [inline keyboards/lily58/r2g/config.h:35]:
```suggestion
```

File: keyboards/lily58/r2g/rules.mk [hunk 2]
```diff
@@ -0,0 +1,2 @@
+LTO_ENABLE = yes
+BOOTLOADER = atmel-dfu
```
Comment (fauxpark) [inline keyboards/lily58/r2g/rules.mk:2]:
```suggestion
```

File: keyboards/lily58/r2g/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+```sh
+make crkbd/r2g:mb_via
+```
+
+Flash example for this keyboard:  
+```sh
+qmk flash -kb crkbd/r2g -km mb_via
+```
```
Comment (fauxpark) [inline keyboards/lily58/r2g/readme.md:18]:
```suggestion

    make crkbd/r2g:default

Flashing example for this keyboard:

    make crkbd/r2g:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/lily58/keymaps/mb_via/config.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+#define TAPPING_FORCE_HOLD
+#define TAPPING_TERM 100
+
+// Underglow
+/*
+#undef RGBLED_NUM
+#define RGBLED_NUM 14    // Number of LEDs
+#define RGBLIGHT_SLEEP
+*/
```
Comment (waffle87) [inline keyboards/lily58/keymaps/mb_via/config.h:40]:
```suggestion
```

File: keyboards/lily58/keymaps/mb_via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */
+
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+//extern uint8_t is_master;
```
Comment (waffle87) [inline keyboards/lily58/keymaps/mb_via/keymap.c:21]:
```suggestion
```

File: keyboards/lily58/r2g/config.h [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+#   define ENABLE_RGB_MATRIX_MULTISPLASH
+#endif
+
+#   define SPLIT_TRANSPORT_MIRROR
+#endif
+
+#ifdef RGBLIGHT_ENABLE
+#   define RGB_MATRIX_MAXIMUM_BRIGHTNESS 120
+#endif
+
```
Comment (waffle87) [inline keyboards/lily58/r2g/config.h:67]:
```suggestion
```

File: keyboards/lily58/r2g/info.json [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+            ]
+        }
+    },
+    "rgblight": {
+        "pin": "D3",
+        "led_count": 74,
+        "split": true,
+        "split_count": [37, 37]
+    },
+
```
Comment (waffle87) [inline keyboards/lily58/r2g/info.json:44]:
```suggestion
```

File: keyboards/lily58/r2g/config.h [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+//#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+//#define LOCKING_RESYNC_ENABLE
+
+
+#ifdef RGB_MATRIX_ENABLE
```
Comment (waffle87) [inline keyboards/lily58/r2g/config.h:36]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
#   define RGB_DI_PIN D3
```

File: keyboards/lily58/r2g/r2g.c [hunk 9]
```diff
@@ -0,0 +1,157 @@
+#include "r2g.h"
```
Comment (waffle87) [inline keyboards/lily58/r2g/r2g.c:1]: This file is missing a license header.

File: keyboards/lily58/r2g/r2g.c [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+            oled_write_ln_P(PSTR("Lower"), false);
+            break;
+        case _RAISE:
+            oled_write_ln_P(PSTR("Raise"), false);
+            break;
+        case _ADJUST:
+            oled_write_ln_P(PSTR("Adjust"), false);
+            break;
+    }
+}
```
Comment (waffle87) [inline keyboards/lily58/r2g/r2g.c:37]:
```suggestion
void oled_render_layer_state_r2g(void) {
    oled_write_P(PSTR("Layer: "), false);
    switch (get_highest_layer(layer_state)) {
        case 0:
            oled_write_ln_P(PSTR("Default"), false);
            break;
        case 1:
            oled_write_ln_P(PSTR("Lower"), false);
            break;
        case 2:
            oled_write_ln_P(PSTR("Raise"), false);
            break;
        case 3:
            oled_write_ln_P(PSTR("Adjust"), false);
            break;
        default:
            oled_write_ln_P(PSTR("Unknown"), false);
    }
}
```

File: keyboards/lily58/r2g/r2g.h [hunk 11]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (waffle87) [inline keyboards/lily58/r2g/r2g.h:1]: This file can probably be removed. Will need to remove its inclusion elsewhere and whatnot of course.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**